tiledb/sm/array_schema/dimension.cc

```diff
@@ -1085,6 +1152,59 @@ void Dimension::set_oob_func() {
   }
 }
 
+void Dimension::set_covered_func() {
+  switch (type_) {
+    case Datatype::INT32:
+      covered_func_ = covered<int32_t>;
```

Comment:
I was wondering why `covered` and `overlap_ratio` had an unused `Dimension* dim` argument, but I see it is so we can perform the function assignment here.

I think the right thing to do is to:
1. Remove the `dim` argument from `covered` and `overlap_ratio`.
2. Remove the `const Dimension* dim` from the `covered_func_` and `overlap_ratio_func_` templated typename. In other words, the type for both will be: `std::function<bool(const Range&, const Range&)>`.
3. Instead of assigning the function, bind it without the implicit 'Dimension' instance. In other words, this line becomes: `covered_func_ = std::bind(&covered<int32_t>, std::placeholders::_1, std::placeholders_2)`.

Note that the current line is equivalent to:
`covered_func_ = std::bind(&covered<int32_t>, this, std::placeholders::_1, std::placeholders_2)`. So all I'm suggesting is that we don't need the `this` argument.

Reply:
I think I just need to remove the `Dimension*` argument from the functions that do not use it and everything should work. Will do.